### PR TITLE
Add XR apis to sdk

### PR DESCRIPTION
**Pull request by sandeep-paliwal**
XR APIs for console sdk
TODO add e2e tests once APIs are live

## Before merge, revert these changes

- [ ] re-add `npm test` (git revert 274dd69)

## How Has This Been Tested?
TODO

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by moritzraho — CHANGES_REQUESTED** [review]
Looking good so far, quite a big change 🙂 

We need two more things to make this complete:
- e2e tests for the new endpoints, you can prepare them now but not run until ext reg is up
- remove old app registry apis
- also remove atlaspolicy apis

**Comment by sandeep-paliwal** [comment]
Thanks @moritzraho for the comments.
I will add the e2e tests.
I guess old registry APIs we should deprecate, and once they are decommissioned, then we should remove them from SDK.
WDYT?

**Comment by moritzraho** [comment]
@sandeep-paliwal good thoughts but no one is using this api in the sdk. We discussed this with @meryllblanchet and @fmeschbe I think it is safe to delete those apis.
+ app registry api will not be kept once moving to ext reg

**Comment by meryllblanchet** [comment]
@sandeep-paliwal @moritzraho these APIs should not be exposed within the library and do not serve any 3rd party extensibility purpose. No deprecation needed IMO.

cc @fmeschbe

File: src/index.js [hunk 1]
```diff
@@ -1227,6 +1275,265 @@ class CoreConsoleAPI {
...
+   *  Get all available extension points
+   *
+   * @param {string} organizationId Organization AMS ID
+   * @param {string} xpId xp ID
+   * @param {object} [options] Query options
+   * @param {number} [options.offset] Offset
+   * @param {number} [options.pageSize] page size
+   * @returns {Promise<Response>} the response
+   */
+  async getAllExtensionPoints (organizationId, xpId, options = {}) {
```
[the remark this answers is not in the document]
Comment (sandeep-paliwal) [reply src/index.js:1316]: `xpId` was a required param as per the shared swagger spec. Anyway there is new spec available and we will need to make changes to SDK as per that.

[the remark this answers is not in the document]
Comment (Himavanth) [reply src/index.js:1316]: Looks like there is no change to the param in this API. We should still make this change IMO.

**Comment by shazron** [comment]
@sandeep-paliwal can you get this change into your PR branch: https://github.com/adobe/aio-lib-console/pull/21/files Jesse put it against master so it can't be merged via Github into this PR

**Comment by shazron** [comment]
Moved to #22. Please review any pending changes due to review in this PR that needs to be migrated to the new PR.
The new PR now exists in the main repo, for publishing permission issues.

**Pull request closed without merging**